### PR TITLE
feat: add request camera permission in calling

### PR DESCRIPTION
**Pull request by alexandreferris**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

[AR-1260 - request camera permission](https://wearezeta.atlassian.net/browse/AR-1260)

### Issues

When receiving an incoming call or on an ongoing call, and the user hasn't given permission to access the camera, when clicking on camera button nothing would happen.

### Solutions

Ask for user permission to use its camera when receiving an incoming call or on an ongoing call.

#### How to Test

- Open the app
- send a message to the caller
- receive an incoming call
- tap on the camera button
- permission prompt should pop up

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-539/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3201924)(3201924)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3201924)(3201924)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/calling/controlButtons/CameraButton.kt [hunk 1]
```diff
@@ -10,22 +14,41 @@ import androidx.compose.runtime.mutableStateOf
...
 import androidx.compose.ui.tooling.preview.Preview
 import com.wire.android.R
 import com.wire.android.ui.theme.wireDimensions
+import com.wire.android.util.extension.checkPermission
 
 @Composable
 fun CameraButton(
     isCameraOn: Boolean = false,
-    onHangUpButtonClicked: () -> Unit
+    cameraPermissionDenied: () -> Unit,
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/calling/controlButtons/CameraButton.kt:28]:
```suggestion
    onCameraPermissionDenied: () -> Unit,
```

What do you think of renaming to the above to follow the pattern we have on other permission arguments ?

**Review by yamilmedina — APPROVED** [review]
💯  lgtm! just a small suggestion for an argument name, that follows the pattern of the permissions we have :D

**Review by yamilmedina — COMMENTED** [review]
Just a small suggestion for an argument name :D other than that looks great! 🚀

**Review by ohassine — APPROVED** [review]
🚀🚀

File: app/src/main/kotlin/com/wire/android/ui/calling/controlButtons/CameraButton.kt (hunk 1 above)
Comment (alexandreferris) [reply app/src/main/kotlin/com/wire/android/ui/calling/controlButtons/CameraButton.kt]: Great catch!

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-539/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3208204)(3208204)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3208204)(3208204)-dev-release.apk)

**Pull request merged**